table/neuron/table.cpp

```diff
@@ -0,0 +1,409 @@
+    };
+    static void _hoc_sigmoid1(void) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        Prop* _local_prop = _prop_id ? _extcall_prop : nullptr;
+        _nrn_mechanism_cache_instance _ml_real{_local_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _local_prop ? _nrn_mechanism_access_dparam(_local_prop) : nullptr;
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        hoc_retpushx(_r);
+    }
+    static double _npy_sigmoid1(Prop* _prop) {
+        double _r{};
+        Datum* _ppvar;
+        Datum* _thread;
+        NrnThread* _nt;
+        _nrn_mechanism_cache_instance _ml_real{_prop};
+        auto* const _ml = &_ml_real;
+        size_t const id{};
+        _ppvar = _nrn_mechanism_access_dparam(_prop);
+        _thread = _extcall_thread.data();
+        _nt = nrn_threads;
+        auto inst = make_instance_tbl(_ml_real);
+        check_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt);
+        _r = 1.;
+        sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, *getarg(1));
+        return(_r);
+    }
+
+
+    inline int f_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, double v) {
+        int ret_f_sigmoid1 = 0;
+        inst.sig[id] = 1.0 / (1.0 + exp(inst.global->k * (v - inst.global->d)));
+        return ret_f_sigmoid1;
+    }
+
+
+    void check_sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt) {
+        if (inst.global->usetable == 0) {
+            return;
+        }
+        static bool make_table = true;
+        static double save_k;
+        static double save_d;
+        if (save_k != inst.global->k) {
+            make_table = true;
+        }
+        if (save_d != inst.global->d) {
+            make_table = true;
+        }
+        if (make_table) {
+            make_table = false;
+            inst.global->tmin_sigmoid1 =  -100.0;
+            double tmax = 100.0;
+            double dx = (tmax-inst.global->tmin_sigmoid1) / 100.;
+            inst.global->mfac_sigmoid1 = 1./dx;
+            double x = inst.global->tmin_sigmoid1;
+            for (std::size_t i = 0; i < 101; x += dx, i++) {
+                f_sigmoid1_tbl(_ml, inst, id, _ppvar, _thread, _nt, x);
+                inst.global->t_sig[i] = inst.sig[id];
+            }
+            save_k = inst.global->k;
+            save_d = inst.global->d;
+        }
+    }
+
+
+    inline int sigmoid1_tbl(_nrn_mechanism_cache_range* _ml, tbl_Instance& inst, size_t id, Datum* _ppvar, Datum* _thread, NrnThread* _nt, double v){
+        if (inst.global->usetable == 0) {
```

Comment:
This puts a lot of faith in the optimizer. If it can't see that `inst.global->usetable` is likely a constant, it can't even attempt to lift this branch outside of a loop. Which I suspect will cause it to fail to vectorize.